### PR TITLE
Added broker and trigger tests to keda reconciler tests

### LINKED ISSUE #3814: Add Broker/Trigger tests to KEDA test job
**Problem**
Currently the KEDA tests only test KafkaSource tests. However, with Broker/Triggers now supporting KEDA scaling as well we should run those tests in the same environment to make sure everything works performantly with KEDA and autoscaling enabled

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?

**Exit Criteria**
All Broker/Trigger reconciler tests run in the KEDA test job

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve? 1

**Additional context (optional)**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by NeerajNagure**
Fixes #3814 

## Proposed Changes

- Added broker and trigger tests to keda reconciler tests so as to test them in keda environment.

- 🎁 Added broker and trigger tests

### CODE REVIEW

File: test/keda-reconciler-tests.sh [hunk 1]
```diff
@@ -45,4 +45,30 @@ if ! ${LOCAL_DEVELOPMENT}; then
...
+
+go_test_e2e -tags=deletecm ./test/e2e_new/... || fail_test "E2E (new deletecm) suite failed"
+
+echo "Running E2E Reconciler tests with consumergroup id template changed"
+
+kubectl apply -f "$(dirname "$0")/config-kafka-features/new-cg-id.yaml"
+
+go_test_e2e -tags=e2e -timeout=15m ./test/e2e_new -run TestTriggerUsesConsumerGroupIDFromTemplate
+
+kubectl apply -f "$(dirname "$0")/config-kafka-features/restore-cg-id.yaml"
```
Comment (pierDipi) [inline test/keda-reconciler-tests.sh:72]: I believe we don't need this test for KEDA testing
```suggestion
```

File: test/keda-reconciler-tests.sh [hunk 2]
```diff
@@ -45,4 +45,30 @@ if ! ${LOCAL_DEVELOPMENT}; then
...
+else
+	echo "BROKER_CLASS is set to '${BROKER_CLASS}'. Running tests for that broker class."
+fi
+
+if [ "${BROKER_CLASS}" == "KafkaNamespaced" ]; then
+	# if flag exists, only test tests that are relevant to namespaced KafkaBroker
+	echo "BROKER_CLASS is set to 'KafkaNamespaced'. Only running the relevant tests."
+	go_test_e2e -timeout=1h ./test/e2e_new/... || fail_test "E2E (new) suite failed"
+	success
+fi
```
Comment (pierDipi) [inline test/keda-reconciler-tests.sh:60]: Same for this check, we will only run those with BROKER_CLASS=Kafka
```suggestion
```

**Review by pierDipi — COMMENTED** [review]
Thanks @NeerajNagure, left a few cleanup comments

**Comment by NeerajNagure** [comment]
@pierDipi I think so the tests that are left now in my changes are broker tests.How can I add the trigger tests then?

**Comment by Cali0707** [comment]
@NeerajNagure don't worry about the failing tests - those are unrelated to your changes. It looks like our CI is having issues this morning

**Comment by NeerajNagure** [comment]
@Cali0707 I have made the changes and removed -run=kafkasource flag

File: test/keda-reconciler-tests.sh [hunk 3]
```diff
@@ -39,10 +39,18 @@ export_logs_continuously
 
 header "Running tests"
 
-go_test_e2e -timeout=1h -run=KafkaSource ./test/e2e_new/... || fail_test "E2E (new) suite failed"
-
 if ! ${LOCAL_DEVELOPMENT}; then
 	go_test_e2e -run=KafkaSource -tags=sacura -timeout=40m ./test/e2e/... || fail_test "E2E sacura tests failed"
 fi
```
Comment (Cali0707) [inline test/keda-reconciler-tests.sh:44]: One more change (sorry I missed this earlier) - could we also move this to after the `BROKER_CLASS` checks and remove the `KafkaSource` flag as well? Thanks!

**Comment by NeerajNagure** [comment]
Only `reconciler-tests-keda_eventing-kafka-broker_main` and `upgrade-tests_eventing-kafka-broker_main` are not passing and rest all other tests are passing.Is there any issue in my changes @Cali0707 ?

**Comment by pierDipi** [comment]
@NeerajNagure @Cali0707 it seems related ? 
```
broker_auth.go:193: no consumergroup template in config-kafka-features
```

**Comment by pierDipi** [comment]
I mean not directly related to the changes but the new tests are failing

**Comment by NeerajNagure** [comment]
@pierDipi @Cali0707 I think so that we are not able to fetch the `triggers-consumergroup-template` from the kubeclient over here [https://github.com/knative-extensions/eventing-kafka-broker/blob/551691cda3489753cbee5fa4a95a73c729de73d6/test/rekt/features/broker_auth.go#L188C3-L195C4](https://github.com/knative-extensions/eventing-kafka-broker/blob/551691cda3489753cbee5fa4a95a73c729de73d6/test/rekt/features/broker_auth.go#L188-L195) which is why we are getting that error

**Comment by NeerajNagure** [comment]
@Cali0707 @pierDipi Shouldn't we add `kubectl apply -f ./test/e2e_new/config/features.yaml` command in keda reconciler tests as its present in reconciler-tests.sh?

**Comment by NeerajNagure** [comment]
Adding that command didn't work

**Comment by NeerajNagure** [comment]
@Cali0707 there is this error in keda tests `broker_auth.go:193: no consumergroup template in config-kafka-features` due to which its failing.I am not able to figure out how to fix it,can I please get some help?

**Comment by pierDipi** [comment]
@Cali0707 can you help @NeerajNagure ?

**Comment by pierDipi** [comment]
Same error
```
broker_auth.go:193: no consumergroup template in config-kafka-features
```

**Comment by Cali0707** [comment]
> Same error

I'll look into it today 👍

**Comment by Cali0707** [comment]
Hey @NeerajNagure I think I found the problem that you are seeing in your tests - the `enable-keda-autoscaling.yaml` file only includes the feature flag for keda, so we are deleting all the other feature flags when we apply it. To fix this, I would recommend copying the values from https://github.com/knative-extensions/eventing-kafka-broker/blob/main/control-plane/config/eventing-kafka-broker/200-controller/100-config-kafka-features.yaml into https://github.com/knative-extensions/eventing-kafka-broker/blob/main/test/keda/enable-keda-autoscaling.yaml 

You'll notice that this will leave you with two keys that seemingly do the same thing:
1. `controller.autoscaler`
2. `controller-autoscaler-keda`

We should remove the `controller.autoscaler` key from the `enable-keda-autoscaling.yaml` file, and switch the value of `controller-autoscaler-keda` from `disabled` to `enabled`

Hopefully this makes sense, if you have any questions please feel free to ask!

**Review by Cali0707 — COMMENTED** [review]
Hey @NeerajNagure sorry for missing this earlier, I think we also need something like https://github.com/knative-extensions/eventing-kafka-broker/blob/abaec811d9acd5523153f8d8a06d8c6efd126794/test/reconciler-tests.sh#L24 before we actually run the tests

**Comment by Cali0707** [comment]
@NeerajNagure looks like these tests caught a bug! Would you be interested in investigating it? If you're not sure, I'm happy to try and investigate and resolve the bug myself as well

**Comment by NeerajNagure** [comment]
@Cali0707 I couldnt understand the bug so you please go forward in investigating the bug

**Comment by Cali0707** [comment]
Hey @NeerajNagure I think my PR linked above should fix the problems here - we just need to merge that into eventing, and then update the dependencies here. If you want to, it would be great if you review the PR in eventing!

**Comment by NeerajNagure** [comment]
@Cali0707 I think so we are getting this error `tracker.go:139: Waiting for eventing-e2e0 to be deleted`. I could not figure this out. Is this something related to deleting something in knative/eventing?

**Pull request closed without merging**